File: algo-expert/python/algo_expert/arrays/easy/non_constructible_change.py
def non_constructible_change(coins: list) -> int:
    sum_coins = 1

    if len(coins) == 0:
        return sum_coins

    sorted_coins = sorted(coins)
    for c in sorted_coins:
        if c > sum_coins:
            return sum_coins
        sum_coins += c
    return sum_coins

File: algo-expert/typescript/src/arrays/easy/NonConstructibleChange.ts
export function nonConstructibleChange(coins: number[]) {
  const totalCoins = coins.length;
  if (totalCoins == 0) {
    return 1;
  }
  coins.sort((a, b) => a - b);
  let change = 0;
  for (let i in coins) {
    if (coins[i] > change + 1) {
      return change + 1;
    }
    change += coins[i];
  }
  return change + 1;
}
